src/main/java/io/redisearch/aggregation/AggregationRequest.java

```diff
@@ -76,6 +78,15 @@ public AggregationRequest apply(String projection, String alias) {
         projections.put(alias, projection);
         return this;
     }
+    
+    public AggregationRequest apply(String projection, String alias,boolean applyProjectionOnQuery) {
```

Comment:
why not break this function into two functions? 

```
    public AggregationRequest queryApply(String projection, String alias) {
  	queryProjections.put(alias, projection);
        return this;
    }
```

Reply:
I felt that keeping the method name as "apply" will be better as the developer will be able to relate to the RediSearch aggregate syntax.

Reply:
these are two different functionalities which accidentally have the same name